vega_sim/scenario/common/agents.py

```diff
@@ -1129,3 +1138,235 @@ def initialise(self, vega: VegaServiceNull):
+            for m in self.vega.all_markets()
+            if m.tradable_instrument.instrument.name == self.market_name
+        ][0]
+
+        self.asset_id = self.vega.find_asset_id(symbol=self.asset_name)
+        self.vega.mint(
+            self.wallet_name,
+            asset=self.asset_id,
+            amount=self.initial_asset_mint,
+        )
+
+        self.pdp = self.vega._market_pos_decimals.get(self.market_id, {})
+        self.mdp = self.vega._market_price_decimals.get(self.market_id, {})
+        self.adp = self.vega._asset_decimals.get(self.asset_id, {})
+        self.vega.wait_for_total_catchup()
+
+    def step(self, vega_state: VegaState):
+        self._collect_prices()
+        signal = self.momentum_func_dict.get(self.momentum_strategy[0])()
+        if signal == 0:
+            return
+
+        trade_side = vega_protos.SIDE_BUY if signal == 1 else vega_protos.SIDE_SELL
+        volume = (
+            self.random_state.poisson(self.buy_intensity)
+            if signal == 1
+            else self.random_state.poisson(self.sell_intensity)
+        )
+        volume *= self.trading_proportion * self.base_order_size
+
+        if volume:
+            if not self.send_limit_order:
+                self.vega.submit_market_order(
+                    trading_wallet=self.wallet_name,
+                    market_id=self.market_id,
+                    side=trade_side,
+                    volume=volume,
+                    wait=False,
+                    fill_or_kill=False,
+                )
+            else:
+                best_bid, best_ask = self.vega.best_prices(market_id=self.market_id)
+                price = (
+                    best_ask + self.offset_levels / 10**self.mdp
+                    if signal == 1
+                    else best_bid - self.offset_levels / 10**self.mdp
+                )
+                self.vega.submit_order(
+                    trading_wallet=self.wallet_name,
+                    market_id=self.market_id,
+                    order_type="TYPE_LIMIT",
+                    time_in_force="TIME_IN_FORCE_IOC",
+                    side=trade_side,
+                    volume=volume,
+                    price=price,
+                    wait=False,
+                )
+
+    def _collect_prices(self):
+        future_price = (
+            int(self.vega.market_data(market_id=self.market_id).mid_price)
+            / 10**self.mdp
+        )
+        self.prices.append(future_price)
+
+    def _MACD(self):
+        macd, signal = MACD(
+            prices=self.prices,
+            fast_period=self.momentum_strategy[1]["fast_period"],
+            slow_period=self.momentum_strategy[1]["slow_period"],
+            signal_period=self.momentum_strategy[1]["signal_period"],
+        )
+        self.indicators.append(macd - signal)
+
+        if len(self.indicators) == 1:
+            return 0
+
+        if self.indicators[-2] < 0 and self.indicators[-1] >= 0:
+            trade_signal = 1
```

Comment:
Can we turn trade_signal into an enum please? Will be a little clearer what each number means